File: Data Structures & Algorithms/non-cyclical-number/submission-1.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = []
        seen.append(n)
        total = 0
        
        while True:
            total = 0
            
            for c in str(n):
                total += int(c) ** 2
            
            if total in seen:
                break
            elif total == 1:
                break

            n = total
            seen.append(n)

        return total == 1

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> seen = new HashMap<>();

        for (int i = 0; i < nums.length; i++){
            int diff = target - nums[i];
            if (seen.get(diff) != null) {

                return new int[] {seen.get(diff), i};
            }
            seen.put(nums[i], i);
        }
        return new int[] {};
    }
}
